zrml/global-disputes/src/lib.rs

```diff
@@ -566,54 +646,116 @@ mod pallet {
-                    |prev_winner_info| {
-                        if amount >= prev_winner_info.outcome_info.outcome_sum {
-                            WinnerInfo::new(outcome.clone(), amount)
+                    GDInfo::new(outcome.clone(), amount),
+                    |mut prev_gd_info| {
+                        if amount >= prev_gd_info.outcome_info.outcome_sum {
+                            prev_gd_info.update_winner(outcome.clone(), amount);
+                            prev_gd_info
                         } else {
-                            prev_winner_info
+                            prev_gd_info
                         }
                     },
                 ));
             });
         }
+
+        fn reward_shared_possession(
+            reward_info: RewardInfoOf<T>,
+            owners: OwnerInfoOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let mut remainder = reward_info.reward;
+            let owners_len = owners.len() as u32;
+            let owners_len_in_balance: BalanceOf<T> = <BalanceOf<T>>::from(owners_len);
+            if let Some(reward_per_each) = reward_info.reward.checked_div(&owners_len_in_balance) {
+                for winner in owners.iter() {
+                    // *Should* always be equal to `reward_per_each`
+                    let reward = remainder.min(reward_per_each);
+                    remainder = remainder.saturating_sub(reward);
+                    // Reward the loosing funds to the winners
+                    let res = T::Currency::transfer(
+                        &reward_info.source,
+                        winner,
+                        reward,
+                        ExistenceRequirement::AllowDeath,
+                    );
+                    // not really much we can do if it fails
+                    debug_assert!(
+                        res.is_ok(),
+                        "Global Disputes: Rewarding a outcome owner failed."
+                    );
+                }
+            } else {
+                log::error!(
+                    "Global Disputes: There should be always at least one owner for a voting \
+                     outcome."
+                );
+                debug_assert!(false);
+            }
+            Self::deposit_event(Event::OutcomeOwnersRewarded {
+                market_id: reward_info.market_id,
+                owners: owners.into_inner(),
+            });
+            Ok((Some(T::WeightInfo::reward_outcome_owner_with_funds(owners_len))).into())
+        }
+
+        fn reward_paid_possession(
+            reward_info: RewardInfoOf<T>,
+            owner: AccountIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let res = T::Currency::transfer(
+                &reward_info.source,
+                &owner,
+                reward_info.reward,
+                ExistenceRequirement::AllowDeath,
+            );
+            debug_assert!(res.is_ok(), "Global Disputes: Rewarding outcome owner failed.");
```

Comment:
In general if we use matching return types we can just return error from T::Currency::transfer() by using ?; instead of ; and then check res to be ok.

Reply:
Returning and error is useful in this case, thanks!

Reply:
6e53282b3d9e3ee8ef73cc13f8645de20811897b